hkmc2/shared/src/test/mlscript/syntax/annotations/Unsupported.mls

```diff
@@ -90,13 +90,3 @@ id(@Log 5)
 //│ ║  l.82: 	@1 + 2 class Qux
 //│ ╙──      	   ^^^
 //│ = 2
-
-:todo
-class Foo with
-  @debug
-  constructor
-    Bar(qax: Str)
-//│ ╔══[PARSE ERROR] Blocks cannot be annotated
-//│ ║  l.98: 	    Bar(qax: Str)
-//│ ╙──      	^^^^
-//│ /!!!\ Uncaught error: scala.NotImplementedError: List() (of class Nil$)
```

Comment:
Why remove the test case?

Reply:
Why couldn't this code work? The code looks fine to me.

Clearly, this code was triggering a path in the compiler that was previously untested. So it's important that we keep the test around! Why would you remove it and revert to having that path untested?

Reply:
The code path is tested. There are around 11 test blocks which reveals this code path IIRC, for example, in `ADTs.mls`. Also, the `UnimplementedError` is not caused by any code introduced in this commit and can be found in all other test blocks revealing this code path.

Reply:
Here is a test block that reveals the same code path: https://github.com/hkust-taco/mlscript/blob/e9e1354cd539ca8c0d40cdc201b18d6068444d95/hkmc2/shared/src/test/mlscript/basics/ADTs.mls#L14-L18

Reply:
I meant that there is no test that tries to annotate a split statement like this. In fact, the test I'd like to see is:
```fs
@debug
val
  x = 1
  y = 2
```

Reply:
Do you mean a test like this: https://github.com/hkust-taco/mlscript/blob/e907ad68197d93d63b00bb53fadac7dbed0c0522/hkmc2/shared/src/test/mlscript/syntax/annotations/Declarations.mls#L117-L122

Reply:
Or do you want me to specifically introduce a test for `val`?

Reply:
Ok that one seems to be enough 👍